Assert availability of libwebrtc

vitorroriz opened the pull request:
#### ae12b87cfbd4e57ec2173bdda10a158e51697d45
<pre>
Assert availability of libwebrtc
<a href="https://bugs.webkit.org/show_bug.cgi?id=274991">https://bugs.webkit.org/show_bug.cgi?id=274991</a>
<a href="https://rdar.apple.com/problem/129080708">rdar://problem/129080708</a>

Reviewed by Alexey Proskuryakov and Brent Fulgham.

If for some reason libwebrtc is not present, we could crash assert immediately instead of crash on a bad access.
Also, we are reverting <a href="https://commits.webkit.org/279586@main">https://commits.webkit.org/279586@main</a> where we would not crash if libwebrtc
would not be available.

* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.cpp:
(WebCore::WebRTCProvider::webRTCAvailable):

Canonical link: <a href="https://commits.webkit.org/279610@main">https://commits.webkit.org/279610@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/953bdd6786b099df596ba59e9a2903d49a5d5a26

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53968 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33340 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6499 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57246 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4695 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/56271 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40849 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4582 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43697 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3097 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56065 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/31552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46692 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24838 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28392 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/4015 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2844 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/50029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4217 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58839 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29145 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/4295 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51113 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30335 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46828 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/50450 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11755 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31286 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30113 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 reviewed (changes requested):
No, on macOS we need to return the value of !!webrtc::setApplicationStatus because it is false in the restore OS.  The iOS simulator change is good, but we may want to add the assertion there.

aproskuryakov commented on Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.cpp:
```diff
@@ -102,10 +102,11 @@ void LibWebRTCProviderCocoa::setActive(bool value)
     return true;
 #else
-    return !!webrtc::setApplicationStatus;
+    ASSERT(!!webrtc::setApplicationStatus);
```
I think that you want the opposite - assert in embedded case, and `return !!webrtc::setApplicationStatus;` on macOS.

Not sure what's up with watchOS and tvOS here.

cdumez replied:
Watch & TV don't have libwebrtc at the moment I believe.

brentfulgham replied:
I suggest:
`ASSERT_WITH_MESSAGE(!!webrtc::setApplicationStatus, "Failed to find or load libwebrtc");`

brentfulgham reviewed (approved):
I think this assertion could be slightly better if we explained why we care that the result of `!!webrtc::setApplicationStatus` is `true`.

vitorroriz committed:
Assert availability of libwebrtc
https://bugs.webkit.org/show_bug.cgi?id=274991
rdar://problem/129080708

Reviewed by Alexey Proskuryakov and Brent Fulgham.

If for some reason libwebrtc is not present, we could crash assert immediately instead of crash on a bad access.
Also, we are reverting https://commits.webkit.org/279586@main where we would not crash if libwebrtc
would not be available.

* Source/WebCore/platform/mediastream/libwebrtc/LibWebRTCProviderCocoa.cpp:
(WebCore::WebRTCProvider::webRTCAvailable):

Canonical link: https://commits.webkit.org/279610@main
